src/main/java/frc/robot/stateManagement/RobotStateManager.java

```diff
@@ -0,0 +1,71 @@
+package frc.robot.stateManagement;
+
+import edu.wpi.first.networktables.BooleanPublisher;
+import edu.wpi.first.networktables.BooleanTopic;
+import edu.wpi.first.networktables.IntegerPublisher;
+import edu.wpi.first.networktables.IntegerSubscriber;
+import edu.wpi.first.networktables.IntegerTopic;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import java.util.Optional;
+
+public class RobotStateManager extends SubsystemBase {
+  // Alliance Color
+  private final IntegerTopic allianceTopic =
+      NetworkTableInstance.getDefault().getIntegerTopic("ALLIANCE");
+  private final IntegerPublisher alliancePublisher = allianceTopic.publish();
+  private boolean checkAllianceColor = true;
```

Comment:
FMS already stores the result of isRedAlliance.  I don't think we need to have another network table for alliance.  

I would expect RobotStateManager to check whether isRedAlliance in initialization, and store that in a private boolean and then return that information using a getter.

Reply:
done